FileShareService.DesktopClient.Core/AuthProvider.cs

```diff
@@ -97,6 +103,16 @@ public async  Task<string> GetToken()
                 authenticationResult = await publicClientApplication.AcquireTokenInteractive(scopes).ExecuteAsync(cancellationSource.Token);
             }        
             RaisePropertyChanged(nameof(CurrentAccessToken));
+
+            if (AuthProvider.TempToken != CurrentAccessToken )
```

Comment:
Please change `TempToken` to a local variable
````
            var tempToken = CurrentAccessToken;
            try
            {
                authenticationResult = await publicClientApplication.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
                    .ExecuteAsync(cancellationSource.Token)
                    .ConfigureAwait(false);
            }
            // There is no access token in the cache, so prompt the user to sign-in.
            catch (MsalUiRequiredException)
            {
                authenticationResult = await publicClientApplication.AcquireTokenInteractive(scopes).ExecuteAsync(cancellationSource.Token);
            }        
            RaisePropertyChanged(nameof(CurrentAccessToken));

            if (tempToken != CurrentAccessToken && tempToken != null)
            {
                logger.LogInformation("Token renewed silently");
            }
````

Reply:
Resolved